torch/_dynamo/variables/builder.py

```diff
@@ -1585,8 +1585,14 @@ def wrap_module(self, value: torch.nn.Module):
                 for p in value.parameters():
                     self.mark_static_input(p, guard=freezing)
 
-                for b in value.buffers():
-                    self.mark_static_input(b, guard=freezing)
+                # nn buffers is frequently overriden; this check validates
+                # the fn has semantics matching nn.Module expectation
+                # which is callable, and returning iterator of Tensor
+                # Otherwise, we will skip trying to modify buffers
+                if callable(value.named_buffers) and isinstance(value.named_buffers(), Iterator):
```

Comment:
A better would be something like this

1) At import time, get the original `named_buffers` function by saving `orig_module_named_buffers = torch.nn.Module.named_buffers`

2) In L1592, check that `value.named_buffers` (you might have to add .__func__ attribute to get the function from the method) is same as `orig_module_named_buffers`. If it is, only then proceed through marking inputs static.

3) Can you please also do the same L1585 for parameters?